woocommerce/woocommerce-android
# Issue/2844 order detail refactor shipment tracking actions

This PR takes care of the 12/13 step of the Order Detail refactoring #2844 .

#### Changes
- Adds the option to add a new shipment tracking.
- Adds the option to delete a new shipment tracking.

#### Notes
- I have split this massive change into multiple small PRs in order to make reviews easier. 
- This PR is to be merged into a feature branch.
- **This PR is in draft till this [Step 11 PR](https://github.com/woocommerce/woocommerce-android/pull/2876) can be merged.**

#### Testing
- Ensure that you have the shipment tracking plugin enabled on your store.
- Click on an order from the Orders tab.
  - Click on the Add tracking button in the Shipment Trackings section.
  - Add a new order shipment tracking and verify that the new tracking is added successfully to the Order detail screen.
- Click on the ellipsis icon of a shipping label item of an order.
  - Click on the **Copy tracking number** button and verify that the tracking number is copied to clipboard.
  - Click on the **Track shipment* button and verify that you are redirected to the tracking website.
  - Click on the **Delete shipment* button and verify that the shipment is removed from the list and an UNOD SnackBar is displayed. Clicking on undo should add the shipment back to the list. Once the SnackBar is dismissed, the shipment should be deleted from the local db.
- Click on an Order from the Orders TAB that has some shipping labels associated with the order. **To create shipping labels, please follow the instructions provided in the Shipping Labels Master Thread - p91TBi-2om**
  - Click on the ellipsis icon beside the shipping label tracking number. 
  - Click on the **Copy tracking number** button and verify that the tracking number is copied to clipboard.
  - Click on the **Track shipment* button and verify that you are redirected to the tracking website. 

#### Screenshots
##### Add shipment Tracking
<img src="https://user-images.githubusercontent.com/22608780/93733719-2e8ff600-fbf4-11ea-8e9b-1b68df9b93fa.gif" width="300"/>. 

##### Delete shipment tracking
<img src="https://user-images.githubusercontent.com/22608780/93733727-36e83100-fbf4-11ea-8ad0-a3c7a9260c2e.gif" width="300"/>. 

Update release notes:
